Bottom: b2b126cf6ece972cf4f4fdbdd615dc013cc01daa
Top:    6a0cf336ab53d181d2759425ef1b494c83f3dac7
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-09-06 01:15:24 +0100

Refresh of hid-gtk-connect-up-the-file-ch

---

diff --git a/src/file.c b/src/file.c
index be7f6a5..d31b157 100644
--- a/src/file.c
+++ b/src/file.c
@@ -358,6 +358,7 @@ SavePCB (char *file)
 
   gui->notify_save_pcb (file, false);
   retcode = WritePipe (file, true);
+  printf ("File changes hitting disk now\n");
   gui->notify_save_pcb (file, true);
 
   return retcode;
diff --git a/src/hid/gtk/gtkhid-main.c b/src/hid/gtk/gtkhid-main.c
index aa33d85..2b1483a 100644
--- a/src/hid/gtk/gtkhid-main.c
+++ b/src/hid/gtk/gtkhid-main.c
@@ -2170,7 +2170,7 @@ hid_gtk_init ()
   ghid_hid.finish_debug_draw        = ghid_finish_debug_draw;
 
   ghid_hid.notify_save_pcb          = ghid_notify_save_pcb;
-  ghid_hid.notify_filename_changed  = ghid_hid_notify_filename_changed;
+  ghid_hid.notify_filename_changed  = ghid_notify_filename_changed;
 
   hid_register_hid (&ghid_hid);
 #include "gtk_lists.h"
diff --git a/src/hid/gtk/gui-top-window.c b/src/hid/gtk/gui-top-window.c
index 4ba598c..1277da0 100644
--- a/src/hid/gtk/gui-top-window.c
+++ b/src/hid/gtk/gui-top-window.c
@@ -411,7 +411,12 @@ ghid_notify_save_pcb (const char *filename, bool done)
    * that is being saved.
    */
   if (strcmp (filename, PCB->Filename) != 0)
-    return;
+    {
+      printf ("Being notified about save - ignoring save to %s\n", filename);
+      return;
+    }
+
+  printf ("Being notified about save to %s\n", filename);
 
   if (!done)
     disconnect_file_change_monitor (ghidgui);
@@ -422,6 +427,7 @@ ghid_notify_save_pcb (const char *filename, bool done)
 void
 ghid_notify_filename_changed (void)
 {
+  printf ("Notified the PCB file is now under a new name\n");
   disconnect_file_change_monitor (ghidgui);
   connect_file_change_monitor (ghidgui);
 }
diff --git a/src/hid/gtk/gui.h b/src/hid/gtk/gui.h
index 5bb8049..af59121 100644
--- a/src/hid/gtk/gui.h
+++ b/src/hid/gtk/gui.h
@@ -339,7 +339,7 @@ int ghid_drc_window_throw_dialog (void);
 /* In gui-top-window.c  */
 void ghid_update_toggle_flags (void);
 void ghid_notify_save_pcb (const char *file, bool done);
-void ghid_noiify_filename_changed (void);
+void ghid_notify_filename_changed (void);
 void ghid_install_accel_groups (GtkWindow *window, GhidGui *gui);
 void ghid_remove_accel_groups (GtkWindow *window, GhidGui *gui);
